### PR TITLE
AsyncRewriter should ignore accessibility in the EE

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @gafter, @AlekseyTs

File: src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs [hunk 1]
```diff
@@ -324,6 +324,15 @@ private Location GetSmallestSourceLocationOrNull(Symbol symbol)
             return result;
         }
 
+        /// <summary>
+        /// Ignore accessibility when resolving well-known type
+        /// members, in particular for generic type arguments.
+        /// </summary>
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs]: Why might we want to do this?

**Comment by amcasey** [comment]
:+1:

**Comment by amcasey** [comment]
LocalRewriter too?

**Comment by AlekseyTs** [comment]
LGTM. Consider adding more unit-tests, it looks like there is only one test and it is for VB.

**Comment by cston** [comment]
Regarding unit tests, there is a corresponding C# test that was passing with the previous C# implementation. Unfortunately, it's difficult to test to the IgnoreAccessibility false case (from the Compiler rather than the EE) since binding will fail before lowering.

File: src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs]: Added "... (e.g.: binding to internal types in the EE)"

**Comment by cston** [comment]
@amcasey Are there cases where LocalRewriter resolves constructed methods?

**Pull request merged**